### PR TITLE
Remove empty string fallback for CLI

### PR DESCRIPTION
**Pull request by orhun**
I think it's not clear what is happening when `cicero` is run without any arguments. e.g.:

```sh
$ cicero

$
```

In this case, looks like an empty string is processed:

https://github.com/eyeplum/cicero-tui/blob/8c5b5c16dae8fc5bbb1406fda347c021ae09a0af/src/cli/input.rs#L66

In order to make it more clear I added a new `Error` type named `MissingInput` and it's thrown if the user does not provide any input:

```sh
$ cicero
Error: An input or `--tui` flag is expected
$
```

So `cicero` only fallbacks to an empty string if the `--tui` flag is given. Otherwise an error is thrown and user is made aware that they have to provide an input.

### CODE REVIEW

**Comment by eyeplum** [comment]
Thanks @orhun! I will review this during this weekend.

File: src/cli/input.rs
Comment (eyeplum) [inline src/cli/input.rs]: This is just my opinion and might be a bit nitpicking, I think since we are already doing pattern matching, maybe we could do it this way:
```
let input_string = match args.value_of(ARGUMENT_VALUE_NAME_INPUT) {
        Some(v) => Ok(v),
        None => {
            if args.is_present(FLAG_NAME_TUI_MODE) {
                Ok("") // Fallback to an empty string
            } else {
                Err(Box::new(Error::MissingInput))
            }
        }
}?;
```

Here we make the `match` expression return a `Result<&str>`, and unwrap it at the end of the expression (or, if the result is an error, we bail and bubble up the error).

Not 100% sure whether `&str` would work, if it does then great (we saved a copy of the string), if it doesn't, then what you have is fine, since this function only runs once per launch.

**Review by eyeplum — APPROVED** [review]
I haven't settled on what's the best behaviour when the user doesn't provide an input (e.g. maybe we could do what we do here, or we could show the help message, or maybe it's actually a valid use case), but I think what you have here is a good step forward (as previously this case was not handled at all).

I've left a comment about code style, feel free to go with the way you think is the best :)

**Comment by eyeplum** [comment]
Looks like previously in the GitHub Action workflows I only specified `on push`, so the actions on this PR may not be able to run (couldn't figure out how to manually run it).

I've changed the workflows to be run `on push and pull request`, however these actions are still waiting to be run.

In case you can't merge this PR, feel free to ping me when you are ready, I can merge it for you :)

File: src/cli/input.rs
Comment (orhun) [reply src/cli/input.rs]: I think this is a good suggestion. I applied these changes in 2556945

**Comment by orhun** [comment]
> Looks like previously in the GitHub Action workflows I only specified `on push`, so the actions on this PR may not be able to run (couldn't figure out how to manually run it).
> 
> I've changed the workflows to be run `on push and pull request`, however these actions are still waiting to be run.

I guess they are running now. Let's see.

> In case you can't merge this PR, feel free to ping me when you are ready, I can merge it for you :)

ping 🐸

**Comment by eyeplum** [comment]
Changes merged, thanks @orhun !

**Pull request merged**
